### ISSUE TITLE
Bring back screen lock

### ISSUE THREAD

**Issue by freefilesync** [osmandapp/Osmand#6770]
A while ago it used to be possible to lock the screen to ignore touch events.

Like I imagine many people, I use Osmand in a bike phone holder. When it's raining, condensation builds up on the inside and makes Osmand think that it randomly is getting screen touches.

Very annoying, as it makes the map dance around. Can you bring back the little padlock that stops this from happening?

**Comment by scaidermern** [comment]
According to #1855 this feature has been dropped several years ago (<= 2016).

**Comment by freefilesync** [comment]
Yeah, bring it back

**Comment by eindgebruiker** [comment]
Yes please. I use Osmand both on my bicycle and on my motorbike. In the rain Osmand sometimes completely disappears from the screen.

**Comment by sonora** [comment]
I think there's apps out there facilitating this and worth looking at e.g. Touch Lock.

**Comment by Rowin63** [comment]
"Touch lock" is fine, but unfortunately it also blocks Osmand from being operated via bluetooth controller (CI controller)

**Comment by habitoti** [comment]
I think rain is actually the biggest issue with touch devices. An iPhone is perfectly fine with getting wet at all, but raindrops keep generating touches on the display and OSMAND starts doing all sorts of crazy things when riding through heavy rain. What would actually be good is to block the whole display and only release again when operating som side button, like e.g. a volume control.
I wouldn't say it's just _nice-to-have_. Right now I found it actually to be unusable while I unfortunately had to ride 2h through heavy rain... :-(

**Comment by schmadde** [comment]
For me this is not only "nice to have" but not having this (any more) is a major showstopper and make the whole app unusable in rainy conditions.

**Comment by MartinVandersteen** [comment]
+1 This happened to me today on my motorcycle, started raining and it's just impossible to keep the navigation working in these conditions without having to take your eyes/hands off the road to readjust. Very necessary !

**Comment by floppy-engineer** [comment]
+1 The app HAS motorcycle/bike profiles but they removed corresponding functionality. Not cool (

**Comment by sonora** [comment]
It's not trivial to package this universally into OsmAnd. Have you tried specialized Apps like Touch Protector?

**Comment by floppy-engineer** [comment]
Nope. I preferred to have only necessary application on my phone. But thanks, I've seen the reference to such type of apps in the tread. I checked such app and I don't like extensive access privileges on it. Maybe it's possible to find a pure app)

**Comment by Jasper-Ben** [comment]
+1

Not having this feature makes osmand useless on the bike the moment there is only the slightest bit of rain involved.

Regarding the forementioned thirdparty app I fully agree with @floppy-engineer. I only run FOSS software on my de-googled phone, I don't want to have to rely on a "shady" closed-source app with extended privileges for this functionality.

**Comment by Sonwon1** [comment]
Yes, please bring this back!

**Comment by readonly24** [comment]
This feature is crucial for all wheater outdoor usage.

**Comment by vshcherb** [comment]
Now there is a feature to lock map to North always that will help partially with random touches

**Comment by scaidermern** [comment]
> Now there is a feature to lock map to North always that will help partially with random touches

The biggest problem with rain is random touch events. Zooming, exiting follow mode, panning, opening random widgets. Rotation events are rather rare :)

**Comment by albertgardu** [comment]
As previous comments, I agree that OsmAnd becomes useless under the rain. This is a must on any GPS navigation device/app

**Comment by Rowin63** [comment]
I also "liked" this issue some time ago, but by now there are alternatives (like the app "touch protector").
Also using an external controller (like the first/old and simple "carpe iter" bluetooth-controller, or something similar) gives you the option to lock the sceen, and beside other nice functions, this works really good.

**Comment by eindgebruiker** [comment]
A 12V operated external bluetooth controller is a nice although expensive option, but won't work on a bicycle.

**Comment by Rowin63** [comment]
> A 12V operated external bluetooth controller is a nice although expensive option, but won't work on a bicycle.

The one I mentioned before has an internal battery.

**Comment by eindgebruiker** [comment]
> The one I mentioned before has an internal battery.

I guess you mean the CI Control, which unfortunately is no longer available.
